src/bot.py

```diff
@@ -75,40 +123,43 @@ def tweet_loop(self, api):
                 else:
                     api.update_status(self.generate_batch())
                     logging.info("Tweeted out a new flower bed!")
-                    logging.info("The next tweet is scheduled to be made in {} minutes".format(cooldown))
+                    logging.info("The next tweet is scheduled to be made in {} seconds".format(cooldown))
                     time.sleep(cooldown)
-            except:
-                print(sys.exc_info())
-                logging.critical("Tweeting failed due to ratelimit. Waiting {} more minutes.".format(cooldown))
-                time.sleep(cooldown)
+                    raise tweepy.TweepError("error")
+            except Exception as ex:
+                exception = type(ex).__name__
                 
+                if exception == 'RateLimitError':
+                    logging.critical("Tweeting failed due to ratelimit. Waiting {} more seconds.".format(cooldown))
+                    time.sleep(cooldown)
+                else:
+                    logging.critical("TweepError not RateLimitError.")
+                    
     def main(self):
         '''
         Main function that takes care of the authentication and the threading.
         '''
+        
         logging.info("Attempting to login!")
-
-
-        # Authenticate to Twitter
         auth = tweepy.OAuthHandler(CONSUMER_KEY, CONSUMER_SECRET)
         auth.set_access_token(ACCESS_KEY, ACCESS_SECRET)
-
         api = tweepy.API(auth)
-
-        try:
-            api.verify_credentials()
-            logging.info("Successfully logged in!")
-            self.tweet_loop(api)
-        except:
-            print("Error during authentication")
-
+        user = api.me()
+        self.user_info(api, user)
+        logging.info("Successfully logged in!")
+        self.tweet_loop(api)
 
 if __name__ == '__main__':
     try:
-        
+        FLAG = True
         TwitterBot = Bot()
-        
         TwitterBot.main()
-    except:
-      print(sys.exc_info())
-      quit()
+        
+    except Exception as ex:
+        exception = type(ex).__name__

```

Comment:
This isn't the Python way of doing this.

Something like [this](https://stackoverflow.com/questions/6095717/python-one-try-multiple-except/6095782) would look way better.

```python
try:
    something()
except SpecificExceptionClass:
    handling1()
except Exception as e:
    handling2()
```

Reply:
Pls explain what is wrong with the syntax. 

`except Exception as e:
       handling()`

Reply:
There's nothing wrong with it, but "Exception" should be used as a fallback, after all the exceptions you can handle.

Using "Exception" and then if... else cases is definitely not clean.

Reply:
I am catching all forms of exceptions, and then I am using If as a mode to check the type of exception. 
it's basically, 
if ex is Tweeperror -------> authentication error
if ex is RateLimitError -------> rate limit error
else --------> other error, crash

Reply:
> I am catching all forms of exceptions, and then I am using If as a mode to check the type of exception.
> it's basically,
> if ex is Tweeperror -------> authentication error
> if ex is RateLimitError -------> rate limit error
> else --------> other error, crash

Yeah, but that's something that you'd probably do in another language, but catching the Exception, storing it as a string and then comparing it in multiple places across the codebase is not the recommended way to do it. The main process is running under the main() function-- if it crashes, ***ALL*** exceptions will have to be handled in the bottom of the package, and not be split for no reason. It makes the code dirtier.

`try... except` blocks in Python are "self-sufficient" in a sense. "Try this, and if this happens, then do that". What you're proposing is something along the lines of "Try this, unless if something happens. Assign what happened in this variable. If that something is this, then do this. If that something is that, then do that.". So, why should we create more blocks? There's already a design that works and is mostly universal across Python packages and programs, so why would we not do that as well? It's essentially a standard.